ecssd_agent.go

```diff
@@ -371,13 +408,17 @@ func main() {
 	}
 	configuration.HostedZoneId = zoneId
 	metadataClient := ec2metadata.New(session.New())
-	hostname, err := metadataClient.GetMetadata("/hostname")
-	configuration.Hostname = hostname
+	localIP, err := metadataClient.GetMetadata("/local-ipv4")
+	configuration.Hostname = "ip-" + strings.Replace(localIP, ".", "-", 3) + "." + DNSName
```

Comment:
So we can look at only the records matching the "ip-a-b-d-c.domain" pattern, but then if someone has a hostname that doesn't match that pattern, they'll need to manually change the lambda function to match the different pattern. 

Regarding the SetIdentifier field, that's a great idea, but unfortunately it's not available for standard A records.
Below is an excerpt from the Route53 Python documentation regarding the SetIdentifier field.(http://boto3.readthedocs.io/en/latest/reference/services/route53.html#Route53.Client.change_resource_record_sets)

>Weighted, Latency, Geo, and Failover resource record sets only: An identifier that differentiates among multiple resource record sets that have the same combination of DNS name and type. The value of SetIdentifier must be unique for each resource record set that has the same combination of DNS name and type. Omit SetIdentifier for any other types of record sets.